price-feeder/oracle/provider/huobi.go

```diff
@@ -157,17 +190,27 @@ func (p *HuobiProvider) messageReceived(messageType int, bz []byte, reconnectTic
 	}
 
 	var tickerResp HuobiTicker
+	var candleResp HuobiCandle
+
 	if err := json.Unmarshal(bz, &tickerResp); err != nil {
 		// sometimes it returns other messages which are not ticker responses
 		p.logger.Err(err).Msg("failed to unmarshal message")
 		return
 	}
 
-	if tickerResp.Tick.LastPrice == 0 {
+	if tickerResp.Tick.LastPrice != 0 {
+		p.setTickerPair(tickerResp)
+		return
+	}
+
+	if err := json.Unmarshal(bz, &candleResp); err != nil {
+		p.logger.Err(err).Msg("failed to unmarshal message")
 		return
 	}
 
-	p.setTickerPair(tickerResp)
+	if candleResp.Tick.Open != 0 {
+		p.setCandlePair(candleResp)
+	}
```

Comment:
Does anything need to be done if the response is neither ticker nor candle? Currently it would return without error but also without setting anything it looks like

Reply:
I don't think so, the original implementation just would return here without an error if there were no ticker to set. 

cc : @RafilxTenfen

Reply:
it depends on the provider to send different messages, on the huobi one I only saw messages sent that we subscribed for